--- a/src/main/java/net/runelite/launcher/Launcher.java
+++ b/src/main/java/net/runelite/launcher/Launcher.java
@@ -228,9 +228,6 @@
 				jvmProps.put("apple.awt.application.appearance", "system");
 			}
 
-			// Stream launcher version
-			jvmProps.put(LauncherProperties.getVersionKey(), LauncherProperties.getVersion());
-
 			if (settings.isSkipTlsVerification())
 			{
 				jvmProps.put("runelite.insecure-skip-tls-verification", "true");
@@ -412,6 +409,8 @@
 			// Set hs_err_pid location. This is a jvm param and can't be set at runtime.
 			log.debug("Setting JVM crash log location to {}", CRASH_FILES);
 			jvmParams.add("-XX:ErrorFile=" + CRASH_FILES.getAbsolutePath());
+			// RuneLite requires assertions when developer mode is enabled.
+			jvmParams.add("-ea");
 			// Add VM args from cli/env
 			jvmParams.addAll(getJvmArgs(settings));
 
@@ -566,6 +565,7 @@
 	private static Collection<String> getClientArgs(LauncherSettings settings)
 	{
 		final var args = new ArrayList<>(settings.clientArguments);
+		args.add("--developer-mode");
 
 		String clientArgs = System.getenv("RUNELITE_ARGS");
 		if (!Strings.isNullOrEmpty(clientArgs))
